lib/shared/src/models/sync.test.ts

```diff
@@ -641,51 +553,146 @@ describe('syncModels', () => {
+            }
+            expect(model.contextWindow.maxInputTokens).toBe(wantMaxInputTokens)
+        }
+    })
+
+    it('preserves the context window for enterprise users with Claude-3-Sonnet (Pro tier) when the feature flag is on', () => {
         const models = [
             {
-                modelRef: 'openai::latest::gpt-4o' as const,
-                modelName: 'gpt-4o',
-                displayName: 'GPT-4o',
+                modelRef: 'anthropic::latest::claude-3-sonnet' as const,
+                modelName: 'claude-3-sonnet',
+                displayName: 'Claude 3 Sonnet',
                 capabilities: ['chat'],
-                category: ModelTag.Balanced as const,
-                status: 'stable' as const,
+                category: ModelTag.Power as const,
+                status: ModelTag.Experimental as const,
                 tier: ModelTag.Pro as const,
                 contextWindow: {
-                    maxInputTokens: 100000,
-                    maxOutputTokens: 8000,
+                    maxInputTokens: 175000,
+                    maxOutputTokens: 4000,
                 },
             },
         ] satisfies ServerModel[]
 
-        const result = maybeAdjustContextWindows(models, false, true, true)
-        expect(result[0].contextWindow.maxInputTokens).toBe(100000)
-        expect(result[0].contextWindow.maxOutputTokens).toBe(8000)
+        const result = maybeAdjustContextWindows(models, {
+            tier: 'enterprise',
+            longContextWindowFlagEnabled: true,
+        })
+        expect(result[0].contextWindow.maxInputTokens).toBe(175000)
+        expect(result[0].contextWindow.maxOutputTokens).toBe(4000)
     })
 
-    it('Flag on, pro users with gpt o1 (Pro tier)', () => {
+    it('reduces the output tokens for Pro users with Gemini-1.5-Pro (Pro tier) when the feature flag is on', () => {
         const models = [
             {
-                modelRef: 'openai::latest::gpt-o1' as const,
-                modelName: 'gpt-o1',
-                displayName: 'GPT-o1',
+                modelRef: 'google::v1::gemini-1.5-pro' as const,
+                modelName: 'gemini-1.5-pro',
+                displayName: 'Gemini 1.5 Pro',
                 capabilities: ['chat'],
-                category: ModelTag.Balanced as const,
-                status: 'stable' as const,
+                category: ModelTag.Power as const,
+                status: ModelTag.Experimental as const,
                 tier: ModelTag.Pro as const,
                 contextWindow: {
                     maxInputTokens: 175000,
-                    maxOutputTokens: 32000,
+                    maxOutputTokens: 8000,
                 },
             },
         ] satisfies ServerModel[]
 
-        const result = maybeAdjustContextWindows(models, true, false, true)
+        const result = maybeAdjustContextWindows(models, {
+            tier: 'pro',
+            longContextWindowFlagEnabled: true,
+        })
```

Comment:
```suggestion
        const es = maybeAdjustContextWindows(models, {
            tier: 'enterprise',
            longContextWindowFlagEnabled: true,
        })
        const pro = maybeAdjustContextWindows(models, {
            tier: 'pro',
            longContextWindowFlagEnabled: true,
        })
        const free = maybeAdjustContextWindows(models, {
            tier: 'free',
            longContextWindowFlagEnabled: true,
        })
```

For next time or in a follow-up, it'd be easier to put all tier in one condition so that'd be easier to test and compare :D